apps/api/src/app/integrations/e2e/update-integration.e2e.ts

```diff
@@ -36,48 +37,126 @@ describe('Update Integration - /integrations/:integrationId (PUT)', function ()
     expect(integration.credentials.secretKey).to.equal(payload.credentials.secretKey);
   });
 
-  it('should deactivate other providers on the same channel', async function () {
-    const firstProviderPayload = {
-      providerId: 'sendgrid',
+  it('should not allow to update the integration with same identifier', async function () {
+    const identifier2 = 'identifier2';
+    const integrationOne = await integrationRepository.create({
+      name: 'Test1',
+      identifier: 'identifier1',
+      providerId: EmailProviderIdEnum.SendGrid,
+      channel: ChannelTypeEnum.EMAIL,
+      active: false,
+      _organizationId: session.organization._id,
+      _environmentId: session.environment._id,
+    });
+    await integrationRepository.create({
+      name: 'Test2',
+      identifier: identifier2,
+      providerId: EmailProviderIdEnum.SendGrid,
+      channel: ChannelTypeEnum.EMAIL,
+      active: false,
+      _organizationId: session.organization._id,
+      _environmentId: session.environment._id,
+    });
+    const payload = {
+      identifier: identifier2,
+      check: false,
+    };
+
+    const { body } = await session.testAgent.put(`/v1/integrations/${integrationOne._id}`).send(payload);
+
+    expect(body.statusCode).to.equal(400);
```

Comment:
As the same reasoning as in my previous comment before this should be a 409 Conflict.